### PR TITLE
Not throwing on negative reputation

### PR DESCRIPTION
**Pull request by leviadam**
This PR changes the logic for decreasing more reputation than available. 
In the past this scenario would revert. Now it will turn the reputation of this agent to zero.
Adapted the tests to the new logic.

In addition I have added call for the event for mint reputation.
Also adjusted the gas limit and the tests accordingly and updated solium version.

### CODE REVIEW

File: contracts/controller/Reputation.sol [hunk 1]
```diff
@@ -51,8 +51,12 @@ contract Reputation is Ownable {
             balances[_to] = balances[_to].add(absAmount);
         } else {
             absAmount = uint((-1)*_amount);
-            totalSupply = totalSupply.sub(absAmount);
-            balances[_to] = balances[_to].sub(absAmount);
+            if (absAmount > balances[_to]) {
+                absAmount = balances[_to];
+                _amount = (-1)*int(balances[_to]);
```
Comment (orenyodfat) [inline contracts/controller/Reputation.sol]: > amount = (-1)*int(balances[_to]);

 do we need this line ?

File: contracts/controller/Reputation.sol [hunk 2]
```diff
@@ -51,8 +51,12 @@ contract Reputation is Ownable {
             balances[_to] = balances[_to].add(absAmount);
```
Comment (orenyodfat) [inline contracts/controller/Reputation.sol]: @leviadam I guess that the use of safe math to protect against an overflow.right ?
So why not just test that here?.if it get to uint MAX_VALUE just set it to that..avoiding the call to safeMath.

Comment (leviadam) [reply contracts/controller/Reputation.sol]: I think using safeMath is cleaner.

**Comment by orenyodfat** [comment]
If there is no constraint on the rep so max rep is max uint .
One should calculate the exact rep to add in order to get to that ...
If there is a constraint so it will be revert on the constraint .

**Comment by leviadam** [comment]
@orenyodfat  I don't think we need such a max rep.

File: docs/README.md [hunk 3]
```diff
@@ -1,7 +1,54 @@
-# DAOStack Documentation
+# Resources
```
Comment (orenyodfat) [inline docs/README.md]: How this file is related ?

File: docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+## An Overview of Arc
```
Comment (orenyodfat) [inline docs/concepts.md]: How this file is related ?

File: docs/README.md (hunk 3 above)
Comment (leviadam) [reply docs/README.md]: I have no idea. I thought I rebased.

**Comment by leviadam** [comment]
Strange, I was sure I rebased. Checking.

**Comment by leviadam** [comment]
OK. Closing this and will PR from a new branch to save all the mess.

**Pull request closed without merging**